src/LanguageServer/Protocol/LanguageInfoProvider.cs

```diff
@@ -43,24 +43,31 @@ internal class LanguageInfoProvider : ILanguageInfoProvider
             { ".mts", s_typeScriptLanguageInformation },
         };
 
-        public LanguageInformation GetLanguageInformation(string documentPath, string? lspLanguageId)
+        public LanguageInformation GetLanguageInformation(Uri uri, string? lspLanguageId)
```

Comment:
Is `lspLanguageId` the language selected in the VS Code status bar for a particular document? If so, why would we not want to honor that over the file extension, in the case that the user has chosen something different?

Reply:
It is the languageId the LSP client gives us (which I believe in VSCode will change based on the language selected in the status bar). 

I suppose generally we should use the languageId instead (if we're able to map it), otherwise fall back to the file extension.

Reply:
I'm going to defer this for now though - there is some small risk of regression, especially in VS if they give us the wrong languageId.  I'd rather take that in a separate PR